### PR TITLE
feat(content-block-media): web-component created

### PR DESCRIPTION
**Pull request by RaphaelAmadeu**
### Related Ticket(s)

#3789 

### Description

Created the web-components version of `ContentBlockMedia`, based on the react version.

### CODE REVIEW

File: packages/web-components/examples/codesandbox/components/content-block-media/src/index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ */
+
+import '@carbon/ibmdotcom-web-components/es/components/content-block-media/content-block-media';
+import '@carbon/ibmdotcom-web-components/es/components/content-block-media/content-block-media-content';
+import '@carbon/ibmdotcom-web-components/es/components/content-block/content-block-heading';
+import '@carbon/ibmdotcom-web-components/es/components/content-block/content-block-paragraph';
+import '@carbon/ibmdotcom-web-components/es/components/content-group/content-group-heading';
+import '@carbon/ibmdotcom-web-components/es/components/content-item/content-item-heading';
+import '@carbon/ibmdotcom-web-components/es/components/content-item/content-item-copy';
+import '@carbon/ibmdotcom-web-components/es/components/content-item/content-item';
```
Comment (asudoh) [inline packages/web-components/examples/codesandbox/components/content-block-media/src/index.js:17]: Some bundlers, including Snowpack, require `.js` extension:

```suggestion
import '@carbon/ibmdotcom-web-components/es/components/content-block-media/content-block-media.js';
import '@carbon/ibmdotcom-web-components/es/components/content-block-media/content-block-media-content.js';
import '@carbon/ibmdotcom-web-components/es/components/content-block/content-block-heading.js';
import '@carbon/ibmdotcom-web-components/es/components/content-block/content-block-paragraph.js';
import '@carbon/ibmdotcom-web-components/es/components/content-group/content-group-heading.js';
import '@carbon/ibmdotcom-web-components/es/components/content-item/content-item-heading.js';
import '@carbon/ibmdotcom-web-components/es/components/content-item/content-item-copy.js';
import '@carbon/ibmdotcom-web-components/es/components/content-item/content-item.js';
```

File: packages/web-components/src/components/content-block-media/__stories__/README.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+```javascript
+import '@carbon/ibmdotcom-web-components/es/components/content-block-media/content-block-media';
+import '@carbon/ibmdotcom-web-components/es/components/content-block-media/content-block-media-content';
+import '@carbon/ibmdotcom-web-components/es/components/content-block/content-block-heading';
+import '@carbon/ibmdotcom-web-components/es/components/content-block/content-block-paragraph';
+import '@carbon/ibmdotcom-web-components/es/components/content-group/content-group-heading';
+import '@carbon/ibmdotcom-web-components/es/components/content-item/content-item-heading';
+import '@carbon/ibmdotcom-web-components/es/components/content-item/content-item-copy';
+import '@carbon/ibmdotcom-web-components/es/components/content-item/content-item';
```
Comment (asudoh) [inline packages/web-components/src/components/content-block-media/__stories__/README.stories.mdx:27]:
```suggestion
import '@carbon/ibmdotcom-web-components/es/components/content-block-media/content-block-media.js';
import '@carbon/ibmdotcom-web-components/es/components/content-block-media/content-block-media-content.js';
import '@carbon/ibmdotcom-web-components/es/components/content-block/content-block-heading.js';
import '@carbon/ibmdotcom-web-components/es/components/content-block/content-block-paragraph.js';
import '@carbon/ibmdotcom-web-components/es/components/content-group/content-group-heading.js';
import '@carbon/ibmdotcom-web-components/es/components/content-item/content-item-heading.js';
import '@carbon/ibmdotcom-web-components/es/components/content-item/content-item-copy.js';
import '@carbon/ibmdotcom-web-components/es/components/content-item/content-item.js';
```

File: packages/web-components/src/components/content-block-media/content-block-media-content.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  //  Overrides the media position in relation to the copy in the content group
+  //  eslint-disable-next-line class-methods-use-this
+  protected _renderContent() {
+    // Renders `<div class="bx--content-item">` directly instead of using `<dds-content-item>`
+    // because `<dds-content-block-simple>` uses only the copy content
+    return html`
+      <div class="${prefix}--content-block__children">
+        <div class="${prefix}--content-block-simple__content">
+          <div>
+            <slot name="media"></slot>
```
Comment (asudoh) [inline packages/web-components/src/components/content-block-media/content-block-media-content.ts:45]: Seems that there are two `<slot>`s with the same `name`?

**Review by asudoh — COMMENTED** [review]
Thank you for creating this @RaphaelAmadeu!

File: packages/web-components/src/components/content-block-media/content-block-media-content.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+ *
+ * @element dds-content-block-media-content
+ */
+@customElement(`${ddsPrefix}-content-block-media-content`)
+class DDSContentBlockMediaContent extends StableSelectorMixin(DDSContentGroupSimple) {
+  //  Overrides the media position in relation to another slotted components
+  //  eslint-disable-next-line class-methods-use-this
+  protected _renderCopy() {
+    return html`
+      <slot name="media"></slot>
```
Comment (asudoh) [inline packages/web-components/src/components/content-block-media/content-block-media-content.ts:31]: Did you want to put `<slot name="media">` to here, or in `_renderContent()`? Note that a shadow DOM should not have two `<slot>`s with the same name.

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @RaphaelAmadeu for the updates!

**Comment by jeffchew** [comment]
Thanks @RaphaelAmadeu ! Can you add stable selectors to the documentation?

**Comment by asudoh** [comment]
@RaphaelAmadeu Would you want to review/update the snapshot? Thanks!

**Pull request merged**